### PR TITLE
Android implementation

### PR DESCRIPTION
**Pull request by euclides-jr**

### CODE REVIEW

File: android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java [hunk 1]
```diff
@@ -165,6 +184,31 @@ protected void onDestroy() {
         super.onDestroy();
     }
 
+    private void showControls() {
+        Boolean autoHide = pluginOptions.getBoolean(AUTO_HIDE_CONTROLS_OPTION, true);
+
+        if (isAnimatingControls || !autoHide) return;
+
+        isAnimatingControls = true;
```
Comment (marton78) [inline android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java]: I think this variable shouldn't exist. Rather, you could use `controls.getVisibility()`.

File: android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java [hunk 2]
```diff
@@ -165,6 +184,31 @@ protected void onDestroy() {
         super.onDestroy();
     }
 
+    private void showControls() {
```
Comment (marton78) [inline android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java]:
```suggestion
    private void showForAFewSeconds() {
```

File: android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java [hunk 3]
```diff
@@ -210,9 +214,33 @@ public void onAnimationEnd(Animator animation) {
...
         hangupActionFab.setOnClickListener(hangupClickListener());
+
+    private int getLayoutGravity(String position) {
+        int result = Gravity.BOTTOM | Gravity.END;
+
+        if (position.equals("bottom_center")) {
+            result = Gravity.BOTTOM | Gravity.CENTER;
+        } else if (position.equals("bottom_start")) {
+            result = Gravity.BOTTOM | Gravity.START;
+        } else if (position.equals("bottom_end")) {
```
Comment (f1shear) [inline android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java:268]: there is no else

File: android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java [hunk 4]
```diff
@@ -230,9 +235,22 @@ private void initializeUI() {
...
+            bigHangupActionFab.setOnClickListener(hangupClickListener());
+
+            bigHangupActionFab.setVisibility(View.VISIBLE);
+            hangupActionFab.setVisibility(View.GONE);
+        } else if (buttonSize.equals("normal")) {
+            hangupActionFab.setOnClickListener(hangupClickListener());
+
+            hangupActionFab.setVisibility(View.VISIBLE);
+            bigHangupActionFab.setVisibility(View.GONE);
+        }
```
Comment (f1shear) [inline android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java:258]: no else case

File: android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java [hunk 5]
```diff
@@ -484,11 +488,16 @@ public void onConnected(Room room) {
             @Override
             public void onReconnecting(@NonNull Room room, @NonNull TwilioException twilioException) {
                 sendTwilioEvent(TWILIO_EVENT_RECONNECTING);
+
+                statusText.setVisibility(View.VISIBLE);
+                statusText.setText(R.string.reconnecting);
```
Comment (marton78) [inline android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java]: For us, it doesn't matter, but for library users, I think it's better to add all strings as plugin parameters. 

1. Otherwise, they have to change the plugin itself to add a new language.
2. It keeps all internationalized strings nicely in the same place (for us, in en.json / de.json)

**Review by marton78 — CHANGES_REQUESTED** [review]
Also, there is a typo:

*  **Redirect** twilio events to webview

File: android/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1,4 +1,3 @@
 <resources>
-    <string name="connecting">Connecting...</string>
-    <string name="reconnecting">Reconnecting...</string>
+    <string name="my_string">Just a simple string</string>
```
Comment (marton78) [inline android/src/main/res/values/strings.xml]: Why is this needed?

**Review by f1shear — APPROVED** [review]
